zproject/urls.py

```diff
@@ -272,8 +272,8 @@
     url(r'^users/me$', rest_dispatch,
         {'GET': 'zerver.views.users.get_profile_backend',
          'DELETE': 'zerver.views.users.deactivate_user_own_backend'}),
-    # PUT is currently used by mobile apps, we intend to remove the PUT version
-    # as soon as possible. POST exists to correct the erroneous use of PUT.
+    url(r'^users/me/buddy$', rest_dispatch,
+        {'PATCH': 'zerver.views.users.update_buddy_list'}),
```

Comment:
I think probably this should be using an endpoint like `PUT /users/me/buddy_list/12345` as the URL for adding (and DELETE for removing), rather than using a `PATCH` endpoint, since we don't have any metadata in the requests.